core/src/main/kotlin/org/jlleitschuh/guice/Module.kt

```diff
@@ -0,0 +1,10 @@
+package org.jlleitschuh.guice
+
+import com.google.inject.Module
+
+/**
+ * Creates a [Module] with the [BinderScope] being configured when [Module.configure]
+ * is called by Guice.
+ */
+fun module(configure: BinderScope.() -> Unit) =
+    Module { binder -> BinderScope(binder).configure() }
```

Comment:
This is the big addition here that actually makes the API in the readme possible.